src/libs/EmojiUtils.ts

```diff
@@ -569,7 +572,57 @@ function getSpacersIndexes(allEmojis: EmojiPickerList): number[] {
     return spacersIndexes;
 }
 
-export type {HeaderIndice, EmojiPickerList, EmojiSpacer, EmojiPickerListItem};
+type TextWithEmoji = {
+    text: string;
+    isEmoji: boolean;
+};
+
+function splitTextWithEmojis(text = ''): TextWithEmoji[] {
+    if (!text) {
+        return [];
+    }
+
+    // The regex needs to be cloned because `exec()` is a stateful operation and maintains the state inside
+    // the regex variable itself, so we must have a independent instance for each function's call.
+    const emojisRegex = new RegExp(CONST.REGEX.EMOJIS, CONST.REGEX.EMOJIS.flags.concat('g'));
```

Comment:
While working on implementation of splitting logic, @fabioh8010 noticed that `g` flag in regex maintain state between `test()` calls that resulted in different outcomes even if input was the same. We decided that it would be a better/safer option to remove `g` flag and add it when function is called (other functions that are using this regex were updated as well) so we can avoid random bugs